sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTarget.java

```diff
@@ -80,4 +147,29 @@ public synchronized AnalyticsTransmissionTarget getTransmissionTarget(String tra
     String getTransmissionTargetToken() {
         return mTransmissionTargetToken;
     }
+
+    @NonNull
+    private String getEnabledPreferenceKey() {
+        return Analytics.getInstance().getEnabledPreferenceKeyPrefix() + mTransmissionTargetToken.split("-")[0];
+    }
+
+    @WorkerThread
+    private boolean isEnabledInStorage() {
+        return StorageHelper.PreferencesStorage.getBoolean(getEnabledPreferenceKey(), true);
+    }
+
+    @WorkerThread
+    private boolean areAncestorsEnabled() {
+        for (AnalyticsTransmissionTarget target = mParentTarget; target != null; target = target.mParentTarget) {
```

Comment:
Adding a comment here so I can request changes.

Reply:
Checking that now.